### PR TITLE
fixed init datastores bug/added legend button-modal

### PR DESCRIPTION
**Pull request by msouff**
init functions were missing for datastoretype and datastore tables. I just added them back from previous code. The main db initializes fine now, before it would initialize but it would not populate the datastoretype and datastore tables.

---------

I really like the new display. So much that I also made the layer content follow the new the button-modal approach. After that, I got rid of the app navigation for the map.html since it is just taking empty space. Here are a few picks of how it looks before you decide to add it.

screen shot 2017-08-02 at 11 57 26 am
screen shot 2017-08-02 at 11 50 48 am
screen shot 2017-08-02 at 11 50 29 am

Hopefully you like it.

### CODE REVIEW

**Comment by snowman2** [comment]
@msouff, can you create a separate branch and PR with commit 31aea2a?

File: tethysapp/streamflow_prediction_tool/public/css/map.css [hunk 1]
```diff
@@ -1,15 +1,16 @@
 /* MAP STYLES */
 #app-content-wrapper #app-content {
     height: 100%;
+    padding-bottom: 0;
```
Comment (snowman2) [inline tethysapp/streamflow_prediction_tool/public/css/map.css:4]: I like the padding, but will leave everything else the same. Would you mind adding this to a separate PR?

**Comment by msouff** [comment]
You were right about deleting the database manually. I tried specifying the names of the parameters and without the __init__ functions and it worked. I just had to run "tethys syncstores streamflow_prediction_tool -rf"

My last commit also changed the table of content to a treeview style and fixed a small bug that would not let one add watershedgroups. This is how it looks now:

screen shot 2017-08-02 at 4 32 00 pm

Hopefully that helps with the convincing :-)

File: tethysapp/streamflow_prediction_tool/controllers_ajax.py [hunk 2]
```diff
@@ -1346,7 +1346,7 @@ def watershed_group_add(request):
         return JsonResponse({ 'error': "A watershed group with the same name." })
 
     #add Watershed Group
-    group = WatershedGroup(watershed_group_name)
+    group = WatershedGroup(name=watershed_group_name)
```
Comment (snowman2) [inline tethysapp/streamflow_prediction_tool/controllers_ajax.py:1349]: This is useful. Can you put this in a separate PR?

**Comment by snowman2** [comment]
@msouff, I think the table of contents is improving. However, I would like some more time to think about it and maybe brainstorm with you some potential ideas to improve the look and functionality of the table of contents.

In the meantime, if you could pull out some of the other fixes in a separate pull request so I can get those updated, that would be great. When you do it, please update to the most recent version and then add the changes. Thanks!

**Comment by msouff** [comment]
See new pull request

**Comment by snowman2** [comment]
@msouff, sorry about this, but I don't like the modal for the legend.
